src/simulators/statevector/qubitvector.hpp

```diff
@@ -1766,18 +1786,17 @@ double QubitVector<data_t>::expval_pauli(const reg_t &qubits,
         throw std::invalid_argument("Invalid Pauli \"" + std::to_string(pauli[N - 1 - i]) + "\".");
     }
   }
+  return std::make_tuple(x_mask, z_mask, num_y, x_max);
+}
 
-  // Special case for only I Paulis
-  if (x_mask + z_mask == 0) {
-    return norm();
-  }
-
+template <typename data_t>
+void compute_phase(uint_t num_y, std::complex<data_t>& phase){
```

Comment:
The way this function is used it would make more sense to have signature
```suggestion
std::complex<data_t> compute_phase(uint_t num_y)
```

Reply:
This is what I originally had but it caused a template-related compilation error I didn't know how to solve so I went around it. Something alone this lines:

```/home/gadial/Work/qiskit-aer/src/simulators/statevector/qubitvector.hpp: In member function ‘void AER::QV::QubitVector<data_t>::apply_pauli(const reg_t&, const string&)’:
/home/gadial/Work/qiskit-aer/src/simulators/statevector/qubitvector.hpp:1880:35: error: no matching function for call to ‘compute_phase(AER::QV::uint_t&)’
 1880 |   auto phase = compute_phase(num_y);
      |                                   ^
/home/gadial/Work/qiskit-aer/src/simulators/statevector/qubitvector.hpp:1793:22: note: candidate: ‘template<class data_t> std::complex<_Tp> AER::QV::compute_phase(AER::QV::uint_t&)’
 1793 | std::complex<data_t> compute_phase(uint_t& num_y){
      |                      ^~~~~~~~~~~~~
/home/gadial/Work/qiskit-aer/src/simulators/statevector/qubitvector.hpp:1793:22: note:   template argument deduction/substitution failed:
/home/gadial/Work/qiskit-aer/src/simulators/statevector/qubitvector.hpp:1880:35: note:   couldn’t deduce template parameter ‘data_t’
```

Reply:
Ah that makes sense. Yeah because the template param is the return you wouldn't be able to use it with auto, you would have to specify the return type `complex<data_t> phase = compute_phase(num_y)`